io.go

```diff
@@ -0,0 +1,190 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package blobloom
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"unicode"
+)
+
+const maxCommentLen = 44
+
+// Dump writes f to w, with an optional comment string, in the binary format
+// that a Loader accepts. It returns the number of bytes written to w.
+//
+// The comment may contain arbitrary text, within the limits layed out by the
+// format description. It can be used to record the hash function to be used
+// with a Filter.
+func Dump(w io.Writer, f *Filter, comment string) (n int64, err error) {
+	switch {
+	case len(f.b) == 0 || f.k == 0:
+		err = errors.New("blobloom: won't dump uninitialized Filter")
+	case len(comment) > maxCommentLen:
+		err = fmt.Errorf("blobloom: comment of length %d too long", len(comment))
+	}
+	if err != nil {
+		return 0, err
+	}
+
+	var buf [64]byte
+	binary.LittleEndian.PutUint32(buf[12:], uint32(len(f.b)-1))
+	binary.LittleEndian.PutUint32(buf[16:], uint32(f.k))
+	copy(buf[20:], comment)
+
+	if _, err = checkComment(buf[20:]); err != nil {
+		return 0, err
+	}
+
+	k, err := w.Write(buf[:])
+	n = int64(k)
+	if err != nil {
+		return n, err
+	}
+
+	for i := range f.b {
+		for j, x := range f.b[i] {
+			binary.LittleEndian.PutUint32(buf[4*j:], x)
+		}
+		k, err = w.Write(buf[:])
+		n += int64(k)
+		if err != nil {
+			break
+		}
+	}
+
+	return n, err
+}
+
+// A Loader reads a Filter from an io.Reader.
+//
+// A Loader accepts the binary format produced by Dump. The format starts
+// with a 64-byte header:
+//   - the string "blobloom", in ASCII;
+//   - a four-byte version number, which must be zero;
+//   - the number of Bloom filter blocks, minus one, as a 32-bit integer;
+//   - the number of hashes, as a 32-bit integer;
+//   - a comment, at most 44 bytes long, which must be a UTF-8 string
+//     containing only printable characters as determined by unicode.IsGraphic.
+//
+// After the header come the 512-bit blocks, divided into sixteen 32-bit limbs.
+// All integers are little-endian.
+type Loader struct {
+	buf [64]byte
+	r   io.Reader
+
+	Comment string // Comment field. Filled in by NewLoader.
+	nblocks uint64
+	nhashes int
+}
+
+// NewLoader parses the format header from r and returns a Loader
+// that can be used to load a Filter from it.
+func NewLoader(r io.Reader) (*Loader, error) {
+	l := &Loader{r: r}
+
+	err := l.fillbuf()
+	if err != nil {
+		return nil, err
+	}
+
+	version := binary.LittleEndian.Uint32(l.buf[8:])
+	l.nblocks = 1 + uint64(binary.LittleEndian.Uint32(l.buf[12:]))
+	l.nhashes = int(binary.LittleEndian.Uint32(l.buf[16:]))
+	comment := l.buf[20:]
+
+	switch {
+	case string(l.buf[:8]) != "blobloom":
+		err = errors.New("blobloom: not a Bloom filter dump")
```

Comment:
No corresponding dump is present for this.
Thus, it's failing. Can you please check this @greatroar

Reply:
Fixed and tested, thanks. I cherry-picked your Equals method onto master as preparation for this PR.